### PR TITLE
feat: channel updates and fixes.

### PR DESCRIPTION
**Pull request by marvinIsSacul**
## Release Notes

- **New Features**
	- Added interactive action button in the `AlertDialog` for easier dismissal of alerts.
	- Enhanced error handling in the `Charts` component with a retry option.
	- Introduced conditional rendering for input adornments based on channel settings in `RequestMatching`.

- **Improvements**
	- Improved validation logic in `AddChannelScreen` and `EditChannelScreen` for primary routes.
	- Streamlined navigation methods in `AddChannelScreen` to enhance user experience.
	- Enhanced user interface by adding tooltips for primary routes in `ChannelRoutes`.

- **Bug Fixes**
	- Adjusted error messages for better clarity during channel creation failures.

- **Chores**
	- Removed several configuration files related to linting, testing, and building in the `sidebar-app`, which may affect project setup.

### CODE REVIEW

File: packages/channels-app/src/screens/add.channel.screen.tsx [hunk 1]
```diff
@@ -71,20 +68,30 @@ function AddChannelScreen() {
...
+
+  const handleAddChannel = () => {
+    const numOfPrimaryRoutes = channel.routes?.filter(
+      route => !!route.primary
+    ).length
+
+    if (numOfPrimaryRoutes !== 1) {
       showAlert(
-        'Error creating a new channel. ' + error?.response?.data,
+        'Channel must have exactly only 1 primary route.',
```
Comment (drizzentic) [inline packages/channels-app/src/screens/add.channel.screen.tsx:88]: Shouldn't we handle this on the add route modal?

File: packages/channels-app/src/screens/edit.channel.screen.tsx [hunk 2]
```diff
@@ -44,6 +44,19 @@ function EditChannelScreen() {
   })
 
   const handleEditChannel = () => {
+    const numOfPrimaryRoutes = channel.routes?.filter(
+      route => !!route.primary
+    ).length
+
+    if (numOfPrimaryRoutes !== 1) {
+      showAlert(
+        'Channel must have exactly only 1 primary route.',
```
Comment (drizzentic) [inline packages/channels-app/src/screens/edit.channel.screen.tsx:53]: The same comment on add should also apply here

**Review by drizzentic — COMMENTED** [review]
LGTM! left a few comments

**Comment by drizzentic** [comment]
@coderabbitai review

File: packages/channels-app/src/screens/add.channel.screen.tsx (hunk 1 above)
Comment (marvinIsSacul) [reply packages/channels-app/src/screens/add.channel.screen.tsx:88]: It's only at this component whereby we have all the information that we need about all of the channels and their respective routes.

At the deeper nested components, we only have access to the 1 individual route that we're modifying and/or adding.

**Review by drizzentic — APPROVED** [review]
LGTM!

**Pull request merged**